tower-discover/src/lib.rs

```diff
@@ -34,7 +35,10 @@ pub trait Discover {
     type Error;
 
     /// Yields the next discovery change set.
-    fn poll(&mut self) -> Poll<Change<Self::Key, Self::Service>, Self::Error>;
+    fn poll(
```

Comment:
We should probably be making all of these poll methods in traits `Pin` sooner rather than later.

Reply:
I am actually leaning to keeping this without Pin. We thought through many api's and in reality, it makes much more sense right now to keep pin out. So far I have yet to run into issues without Pin.

When I first converted this crate, I had pin but quickly found it to be pretty useless.